src/java/picard/analysis/artifacts/CollectSequencingArtifactMetrics.java

```diff
@@ -196,6 +199,12 @@ protected void acceptRead(final SAMRecord rec, final ReferenceSequence ref) {
             throw new PicardException("Record contains library that is missing from header: " + library);
         }
 
+        // get the ArtifactCounter for this library
+        final ArtifactCounter counter = artifactCounters.get(library);
+        final byte[] readBases = rec.getReadBases();
+        final byte[] readQuals = (USE_OQ && rec.getOriginalBaseQualities() != null) ? rec.getOriginalBaseQualities() : rec.getBaseQualities();
```

Comment:
FYI `rec.getOriginalBaseQualities()` does *not* cache, and so does the attribute lookup and decode from text on each call.  You probably want to replace this with something like:
```java
final byte[] readQuals;
if (USE_OQ) {
    byte[] tmp = rec.getOriginalBaseQualities();
    readQuals = tmp == null ? rec.getBaseQualities() : tmp;
else {
    readQuals = rec.getOriginalBaseQualities()
}
```

Might even be nice to have something like `SAMRecord.getOriginalQualsOrQuals()` which encapsulates that logic?